bin/daemon.php

```diff
@@ -62,41 +63,45 @@
 	die("Unexpected script behaviour. This message should never occur.\n");
 }
 
-$pid = @file_get_contents($pidfile);
+$id = @file_get_contents($pidfile);
 
-if (empty($pid) && in_array($mode, ["stop", "status"])) {
+if (empty($id) && in_array($mode, ["stop", "status"])) {
 	Config::set('system', 'worker_daemon_mode', false);
 	die("Pidfile wasn't found. Is the daemon running?\n");
 }
 
 if ($mode == "status") {
-	if (posix_kill($pid, 0)) {
-		die("Daemon process $pid is running.\n");
+	if (Process::isRunning($id)) {
+		die("Daemon process '" . $id . "' is running.\n");
 	}
 
 	unlink($pidfile);
 
 	Config::set('system', 'worker_daemon_mode', false);
-	die("Daemon process $pid isn't running.\n");
+	die("Daemon process '" . $id . "'  isn't running.\n");
 }
 
 if ($mode == "stop") {
-	posix_kill($pid, SIGTERM);
+	Process::stop($id);
 
 	unlink($pidfile);
 
-	logger("Worker daemon process $pid was killed.", LOGGER_DEBUG);
+	logger("Worker daemon process '" . $id . "' was killed.", LOGGER_DEBUG);
 
 	Config::set('system', 'worker_daemon_mode', false);
-	die("Worker daemon process $pid was killed.\n");
+	die("Worker daemon process '" . $id . "' was killed.\n");
 }
 
-if (!empty($pid) && posix_kill($pid, 0)) {
-	die("Daemon process $pid is already running.\n");
+if (Process::isRunning($id)) {
+	die("Daemon process '" . $id . "' is already running.\n");
 }
 
 logger('Starting worker daemon.', LOGGER_DEBUG);
 
+$trace = debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS, 1);
+$command = basename($trace[0]['file']);
```

Comment:
I had it from here
https://github.com/friendica/friendica/blob/d3a2ed85fe425cffc103030a6027c9b91f9afc69/src/Core/Worker.php#L1127-L1129
But if `__FILE__` is a easier way to find it, I'll use it :-)

Reply:
Without meaning any disrespect for my predecessors, you probably shouldn't trust any code you found elsewhere in this project, 😅

Reply:
Beyond that, this specific trace extraction was meant to work from the method call (looking one stack trace level above), and probably won't work as much from the `bin/daemon.php` file itself. It all depends on what you're expecting to get.

Reply:
Ah OK, I see. That's why I asked for feedback - especially for PHP specific behaviors :-)

I thought using existing functions may be better rather than refinding the wheel, but I will become more critical to the codebase ;-)